fs/layer/node.go

```diff
@@ -823,16 +825,27 @@ func fileModeToSystemMode(m os.FileMode) uint32 {
 	return res
 }
 
-func defaultStatfs(stat *fuse.StatfsOut) {
+func getNameMaxLength() uint32 {
+	if rootPath == "" {
+		return 1<<32 - 1
+	}
+
+	fs := syscall.Statfs_t{}
+	if err := syscall.Statfs(rootPath, &fs); err != nil {
+		return 1<<32 - 1
+	}
+	return uint32(fs.Namelen)
+}
 
+func defaultStatfs(stat *fuse.StatfsOut, nameLen uint32) {
 	// http://man7.org/linux/man-pages/man2/statfs.2.html
 	stat.Blocks = 0 // dummy
 	stat.Bfree = 0
 	stat.Bavail = 0
 	stat.Files = 0 // dummy
 	stat.Ffree = 0
 	stat.Bsize = blockSize
-	stat.NameLen = 1<<32 - 1
+	stat.NameLen = nameLen
```

Comment:
Any reason we can't just make it 255 bytes as same as other filesystems such as ext4? https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits